src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.certificate.infra;
+
+import com.aws.greengrass.clientdevices.auth.api.UseCases;
+import com.aws.greengrass.clientdevices.auth.infra.NetworkState;
+import com.aws.greengrass.clientdevices.auth.iot.CertificateRegistry;
+import com.aws.greengrass.clientdevices.auth.iot.usecases.VerifyIotCertificate;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.inject.Inject;
+
+/**
+ * Periodically calls the VerifyIotCertificate to update the
+ * locally cached certificates on a cadence.
+ */
+public class BackgroundCertificateRefresh implements Runnable {
+    private final CertificateRegistry registry;
+    private final UseCases useCases;
+    private final NetworkState networkState;
+    private static final int DEFAULT_INTERVAL_SECONDS = 60 * 60; // 1H
```

Comment:
This default might be changed. This will probably work with the configuration stuff @vaibhavmurkute is working on.